source/includes/fundamentals/code-snippets/bson-extra-elements/BsonExtraElementsExample.java

```diff
@@ -0,0 +1,74 @@
+package fundamentals;
+
+import static com.mongodb.MongoClientSettings.getDefaultCodecRegistry;
+import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.codecs.pojo.ClassModel;
+import org.bson.codecs.pojo.Conventions;
+import org.bson.codecs.pojo.PojoCodecProvider;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Updates;
+import com.mongodb.client.result.UpdateResult;
+
+public class BsonExtraElementsExample {
+
+	private static String DOC_NAME_VALUE = "MDB0123";
+
+    private static CodecRegistry getCodecRegistry() {
+        ClassModel<ProductWithBsonExtraElements> classModel = ClassModel.builder(ProductWithBsonExtraElements.class).
+                conventions(Arrays.asList(Conventions.ANNOTATION_CONVENTION)).build();
+        PojoCodecProvider pojoCodecProvider = PojoCodecProvider.builder().register(classModel).build();
+        return fromRegistries(getDefaultCodecRegistry(), fromProviders(pojoCodecProvider));
+
+    }
+
+    private static <T> void setup(MongoCollection<T> collection) {
+    	collection.drop();
+
+    	ProductWithBsonExtraElements doc = new ProductWithBsonExtraElements(DOC_NAME_VALUE);
+    	collection.insertOne((T) doc);
+    }
+
+    // Update without the POJO; match the annotations for field names
+    private static void addOtherFields(MongoClient client) {
+    	MongoCollection collection = client.getDatabase("sample_pojo").getCollection("ProductsWithMoreInfo");
+
+    	UpdateResult result = collection.updateOne(
+    			Filters.eq("modelName", DOC_NAME_VALUE),
+    			Updates.combine(
+    					Updates.set("dimensions", "3x4x5"),
+    					Updates.set("weight", "256g")));
+    	System.out.println(result);
+
+    }
+
+    private static <T> void printDocuments(MongoCollection<T> collection) {
+    	 List<T> products = new ArrayList<T>();
```

Comment:
Can be shortened to just:

    collection.find().into(new ArrayList<>()).forEach(System.out::println);

Reply:
Is there any disadvantage to keeping it as is? I was thinking that for writer testing purposes, keeping it as a lambda could allow them to insert debugging/testing logic more easily.

Reply:
Just less idiomatic Java but it's ok.